korge/src/commonMain/kotlin/com/soywiz/korge/view/filter/FlagFilter.kt

```diff
@@ -0,0 +1,75 @@
+package com.soywiz.korge.view.filter
+
+import com.soywiz.klock.TimeSpan
+import com.soywiz.klock.seconds
+import com.soywiz.kmem.toIntCeil
+import com.soywiz.korag.shader.FragmentShader
+import com.soywiz.korag.shader.Uniform
+import com.soywiz.korag.shader.VarType
+import com.soywiz.korag.shader.storageFor
+import com.soywiz.korge.debug.uiEditableValue
+import com.soywiz.korge.view.Views
+import com.soywiz.korui.UiContainer
+import kotlin.math.PI
+import kotlin.math.absoluteValue
+
+
+/**
+ * A Flag [Filter] that distorts the texture using increasing waves to the right, keeping the left-most vertical static,
+ * as if connected to a flag pole.
+ *
+ * [amplitude] is the maximum y amplitude of the waves.
+ * [crestCount] is the number of crests of waves on the x axis.
+ * [cyclesPerSecond] is the number of times the animation would repeat over a second
+ * [time] is the elapsed time of the animation
+ */
+class FlagFilter(
+    amplitude: Double = 80.0,
+    crestCount: Double = 5.0,
+    cyclesPerSecond: Double = 2.0,
+    time: TimeSpan = 0.seconds
+) : ShaderFilter() {
+    companion object {
+        val u_amplitude = Uniform("amplitude", VarType.Float1)
+        val u_crestCount = Uniform("crestCount", VarType.Float1)
+        val u_cyclesPerSecond = Uniform("cyclesPerSecond", VarType.Float1)
+        val u_Time = Uniform("time", VarType.Float1)
+
+        private val FRAGMENT_SHADER = FragmentShader {
+            apply {
+                val x01 = fragmentCoords01.x - ceil(abs(u_amplitude)) / (u_TextureSize.x + 2.lit * ceil(abs(u_amplitude)))
```

Comment:
Please, change `2.lit` with `2f.lit`.

Rationale:
Unfortunately it seems that some opengl drivers support automatic type casting, others not, and this is not checking yet at the DSL level (I want to make the DSL type-safe and perform verifications after building)

![2021-03-03 (3)](https://user-images.githubusercontent.com/570848/109849942-3f886d00-7c52-11eb-8323-c29ba5f1ff93.png)

Reply:
Done